database/scripts/schema_converter_migrate.py

```diff
@@ -0,0 +1,197 @@
+"""
+Script to cut over data from the schema converter cache to the upgraded ukrdc 
+database. 
+"""
+
+from sqlalchemy import create_engine, text, select
+from sqlalchemy.orm import sessionmaker
+from ukrdc_sqla.ukrdc import PatientRecord, PatientNumber, Assessment, DialysisPrescription
+
+def map_id_to_pid(ukrdc_session, cache_conn):
+    """Function to match the patient id in the cache to the patientrecords in
+    the ukrdc. 
+
+    Args:
+        ukrdc_session (_type_): _description_
+        cache_conn (_type_): _description_
+
+    Raises:
+        Exception: _description_
+    """
+
+    # Query to get patient records from cache
+    cache_query = text("""
+        SELECT id, patientid, numbertype, organization 
+        FROM patient;
+    """)
+
+
+    cache_results = cache_conn.execute(cache_query)
+
+    id_lookups = {}  # Dictionary to store mappings
+
+
+    for row in cache_results:
+        # make a mapping between converter patient ids and pids
+        cache_id = row[0]
+        patient_number = row[1]
+        number_type = row[2]
+        organization = row[3]
+
+        query = (
+            select(PatientNumber)
+            .join(PatientRecord, PatientRecord.pid == PatientNumber.pid)
+            .where(
+                PatientNumber.patientid == patient_number,
+                PatientNumber.numbertype == number_type,
+                PatientNumber.organization == organization,
+                PatientRecord.sendingfacility == SENDING_FACILITY,
+                PatientRecord.sendingextract == "UKRDC"
+            )
+        )
+
+        results = ukrdc_session.execute(query).scalars().all()
+        if len(results) >1:
+            raise Exception("Ambiguous match in converter")
+        elif len(results) == 1:
+            id_lookups[cache_id] = results[0].pid
+        else:
+            id_lookups[cache_id] = None
+
+    return id_lookups
+
+
+def map_assessments(ukrdc_session, cache_conn, id_map):
+    """Function to map between the assessments table in the cache and ukrdc
+    """
+
+    #grab assessments  
+    # Query to get patient records from cache
+    query = text("""
+        SELECT * 
+        FROM assessment
+        ORDER BY update_date ASC;             
+    """)
+
+
+    results = [result for result in cache_conn.execute(query)]
+    for result in results:
+        pid = id_map.get(result[0])
+        if pid is not None:
+            # construct compound key since we have lost the order they arrived
+            # in we may have to have a slightly different key
```

Comment:
This would have issues if there were two assessment records with the start date.

I would use the assessment.id field from the converter database in place of assessment.idx in the UKRDC one.